integration/admin_api_test.go

```diff
@@ -252,6 +259,73 @@ func TestCreateAdmin(t *testing.T) {
 	}
 }
 
+func TestCreateClient(t *testing.T) {
+	tests := []struct {
+		client    oidc.ClientMetadata
+		wantError bool
+	}{
+		{
+			client:    oidc.ClientMetadata{},
+			wantError: true,
+		},
+		{
+			client: oidc.ClientMetadata{
+				RedirectURIs: []url.URL{
+					{Scheme: "https", Host: "auth.example.com", Path: "/"},
+				},
+			},
+		},
+	}
+
+	for i, tt := range tests {
+		err := func() error {
+			f := makeAdminAPITestFixtures()
+			s := httptest.NewServer(f.adSrv.HTTPHandler())
+			defer s.Close()
+
+			req := struct {
+				Client  oidc.ClientMetadata `json:"client"`
+				IsAdmin bool                `json:"isAdmin"`
+			}{tt.client, true}
+
+			data, err := json.Marshal(&req)
+			if err != nil {
+				return err
+			}
+			resp, err := f.hc.Post(s.URL+"/api/v1/client", "application/json", bytes.NewReader(data))
```

Comment:
use the admin client to test - that way you know you've implemented the API correctly. See TestGetState for example.